Add update-symlink to data download docs

sjspielman opened the pull request:
Closes #523

This PR updates the documentation for the data download script to also cover the `--update-symlink` flag (note https://github.com/AlexsLemonade/OpenScPCA-analysis/issues/523#issuecomment-2217798972 about why the target branch here is not `main`). 

I added docs for `--update-symlink` in areas where I thought it made sense, and I also bullet-ified some of the instructions for improved legibility while I was here. I have two questions for reviewers:
- Any other spots to add `--update-symlink` to?
- I'm wondering if we actually don't want to keep the docs changes as I've written here, but instead add a section just to describe the symlink situation more explicitly, both in terms of how the scripts behave and in terms of using this new flag. At first I wanted to go in that direction, but then couldn't exactly decide where that section should go. Very curious to hear high-level thoughts about doc organization!
  - That said, this might be good enough for now, and we could reorganize later if contributors frequently have questions or trouble with this behavior.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -16,7 +16,25 @@ Broadly speaking, there are three kinds of ScPCA data you might wish to work wit
-## Accessing ScPCA Data
+
+!!! tip "Log into AWS CLI before running the download scripts"
+    Below, we describe how to use both the `data-download.py` and `results-download.py` script to download ScPCA data and result files, respectively.
```
```suggestion
    Below, we describe how to use the `data-download.py` and `results-download.py` scripts to download ScPCA data and result files, respectively.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -16,7 +16,25 @@ Broadly speaking, there are three kinds of ScPCA data you might wish to work wit
+
+!!! tip "Log into AWS CLI before running the download scripts"
+    Below, we describe how to use both the `data-download.py` and `results-download.py` script to download ScPCA data and result files, respectively.
+    Before running either of these scripts locally (e.g., not from a [virtual computer](../../aws/index.md#lightsail-for-research-virtual-computing-with-aws)), you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
```
I want to clarify that not all virtual computers don't require login; they might have their own virtual systems! So maybe we want to say lightsail specifically as well...

We also probably should say somewhere that downloading test data doesn't require logging in. 

```suggestion
    Before running either of these scripts locally (i.e., not from an ALSF-provided [virtual computer](../../aws/index.md#lightsail-for-research-virtual-computing-with-aws)), you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -16,7 +16,25 @@ Broadly speaking, there are three kinds of ScPCA data you might wish to work wit
 
-## Accessing ScPCA Data
+
+!!! tip "Log into AWS CLI before running the download scripts"
```
I think this tip is in the wrong place now. It should be under the S3 section, as you don't need these scripts for access through the Portal.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -54,62 +72,76 @@ See our documentation [getting access to AWS](index.md#getting-access-to-aws) fo
+    ./download-data.py --dryrun
+    ```
+
+- If you're only working with a subset of the data, you can use the `--projects` or `--samples` to download select projects or samples, respectively (note that these options are mutually exclusive):
````
```suggestion
- If you're only working with a subset of the data, you can use the `--projects` or `--samples` options to download select projects or samples, respectively (note that these options are mutually exclusive)
  - To download multiple projects or samples, provide the ids as a comma separated list
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -54,62 +72,76 @@ See our documentation [getting access to AWS](index.md#getting-access-to-aws) fo
+
+- To update the `data/current` symlink, for example if you have downloaded multiple releases and/or [test data](#accessing-test-data), use the `--update-symlink` flag in one of the following ways.
+  - Note that no data will be downloaded if you use this flag.
+    ```sh
````
Blank line for form. I have some fear that including the blank line before the code block that I would prefer could break the nesting, or I would have suggested it everywhere. But if you can, I would do it everywhere. 
```suggestion

    ```sh
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -54,62 +72,76 @@ See our documentation [getting access to AWS](index.md#getting-access-to-aws) fo
+    ./download-data.py --samples SCPCSXXXXXX,SCPCSXXXXXY
+    ```
+
+- To update the `data/current` symlink, for example if you have downloaded multiple releases and/or [test data](#accessing-test-data), use the `--update-symlink` flag in one of the following ways.
````
I think breaking this up makes sense. I don't necessarily love my rewrite though.
```suggestion
- To change the release that is linked to the `data/current` location use the `--update-symlink` flag.
  - This is useful for switching among multiple previously downloaded releases and/or [test data](#accessing-test-data)
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -54,62 +72,76 @@ See our documentation [getting access to AWS](index.md#getting-access-to-aws) fo
-```sh
-# use the --projects option
-./download-data.py --projects SCPCPXXXXXX
+- To review what samples would be downloaded without performing the download yet, you can use the `--dryrun` option as follows.
````
I think it might make sense to put this option after the projects and maybe show it combined with other options as well. 
```suggestion
- To review the files would be downloaded without performing the download yet, you can use the `--dryrun` option as follows.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -153,60 +185,41 @@ We have provided a [`download-results.py` script](https://github.com/AlexsLemona
-# Get results from a single module
-./download-results.py --modules module-name
+- To download results for one or more modules, use the `--modules` option as follows:
+    ```sh
````
blank space
```suggestion

    ```sh
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -153,60 +185,41 @@ We have provided a [`download-results.py` script](https://github.com/AlexsLemona
-```sh
-# Get results from a single module
-./download-results.py --modules module-name
+- To download results for one or more modules, use the `--modules` option as follows:
````
Here too, I think it is better to be explicit
```suggestion
- To download results for one or more modules, use the `--modules` option with the module directory name(s) as a comma separated list:
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -262,31 +274,28 @@ Similarly, you use the [download result script](#accessing-scpca-module-results)
-TODO: This section needs to be reworked to avoid unexpected downloads. We should recommend --use-release instead.
 
-These commands will download either test data or results, respectively, and will update the `data/current` symlink to instead point to the `data/test` directory.
+Running either of the above commands will update the `data/current` symlink to point to the `data/test` directory, rather than a given data release directory.
```
I think the second clause is more confusing here. (what is "given", where?)
```suggestion
Running either of the above commands will update the `data/current` symlink to point to the `data/test` directory.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -54,62 +72,76 @@ See our documentation [getting access to AWS](index.md#getting-access-to-aws) fo
 
 ```sh
-./download-data.py --dryrun
+./download-data.py --release current
+
+./download-results.py --release current
````
I think it is time to revive this section?

sjspielman commented:
I've gone on a bit of a journey with revisions here! I implemented revisions based on reviews, and I also wound up reorganizing the document a bit (and updated other anchor links across the docs accordingly) to first show the download scripts, before talking about getting data directly from the portal. I also ended up creating a stand-alone section to discuss the symlink behavior, and tried to more strongly emphasize how `data/current` is used in the sections about file structure. 

You'll also notice a big old `TODO` about an admonition that I wonder if we still need.

Let me know how this looks to you now! I do think the document flows a bit more clearly with the changes I implemented, but I can certainly be convinced to back off on some of the updates.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
-./download-data.py --format AnnData
-```
+- To download a specific release other than the most recent release, you can use the `--release` option as follows.
+Again, please download full data releases with caution, as they are quite large!
````
The please makes it sound like this is for our sake, not for theirs.
```suggestion
Again, download full data releases with caution, as they are quite large!
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
-```sh
-./download-data.py --format AnnData
-```
+- To download a specific release other than the most recent release, you can use the `--release` option as follows.
````
I don't feel like "as follows" is needed everywhere, as it becomes pretty self-evident.
```suggestion
- To download a specific release other than the most recent release, you can use the `--release` option.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
-```
+- To download a specific release other than the most recent release, you can use the `--release` option as follows.
+Again, please download full data releases with caution, as they are quite large!
+    - First, you may wish to use the `--list-releases` flag to see all available releases, which are named in `YYYY-MM-DD` format based on their release date.
````
```suggestion
    - You can use the `--list-releases` flag to see all available releases, which are named in `YYYY-MM-DD` format based on their release date.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
-```sh
-# use the --projects option
-./download-data.py --projects SCPCPXXXXXX
+- If you're only working with a subset of the data, you can use the `--projects` or `--samples` option to download select projects or samples, respectively (note that these options are mutually exclusive):
````
Let's move the project/sample options before the release option? It will flow better into the symlink update option mentioned later as well.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
-#### Download data structure
 
-Downloads via the download script will generally have the following structure:
+### Download data file structure
```
Maybe?
```suggestion
### Downloaded data file structure
```

sjspielman replied:
yeah, i agree

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
+    ./download-data.py --projects SCPCPXXXXXX --dryrun
+    ```
+
+The `download-data.py` script will always update the `data/current` symlink to point to the specified release directory.
+You can use the [`--update-symlink` flag described below](#updating-the-current-symlink) to update which release directory the `data/current` symlink directs to.
````
This is not correct. It does _not_ update for specified releases, only when downloading the default `current` release or test data, see https://github.com/jashapiro/OpenScPCA-analysis/blob/e25539c402262ce7123791bc861703c191c1f067/download-data.py#L531-L533

```suggestion
The `download-data.py` script will update the `data/current` symlink to point to the current release directory when run without a `--release` option or to the test data directory when run with `--test-data`.
If you want to instead point to a specific release, you can use the [`--update-symlink` flag described below](#updating-the-current-symlink) to update which release directory the `data/current` symlink directs to.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -152,16 +163,11 @@ If you wish to use output from an `OpenScPCA-nf` analysis module in your module,
 !!! tip "Log into AWS CLI before running the script"
-    Before running this script locally, you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
-    To do this, run the following commands in terminal before running the `results-download.py` script, and follow instructions to log in.
+    Before running this script locally (i.e., not from an [ALSF-provided virtual computer](../../aws/index.md#lightsail-for-research-virtual-computing-with-aws)), you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
```
modify this if you follow the earlier suggestion

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -174,42 +180,49 @@ You can list all the options for `download-results.py` by running the following
     Note that you can also [add this profile definition to your shell profile file](../../technical-setup/environment-setup/configure-aws-cli.md#storing-your-aws-profile-name) so that it will always be defined in any terminal session.
 
-To download results for one or more modules, use the `--modules` option as follows:
+- You can list all possible arguments and their usage for the results download script with the following:
```
```suggestion
- You can list all available arguments and their usage for the results download script with the following:
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -71,49 +59,71 @@ The [`download-data.py` script](https://github.com/AlexsLemonade/OpenScPCA-analy
     Note that you can also [add this profile definition to your shell profile file](../../technical-setup/environment-setup/configure-aws-cli.md#storing-your-aws-profile-name) so that it will always be defined in any terminal session.
 
-We briefly cover some of the most common use cases below, but we encourage you to review all the options available to you.
+- You can list all possible arguments and their usage for the download data script with the following:
```
```suggestion
- You can list all available arguments and their usage for the download data script with the following:
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -174,42 +180,49 @@ You can list all the options for `download-results.py` by running the following
 
 
-#### Download result data structure
+### Download result file structure
```
similar my "maybe?" above
```suggestion
### Downloaded results file structure
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -247,46 +265,69 @@ To list all available releases, you can use the following command from the root
+Running either of the above commands will update the `data/current` symlink to point to the `data/test` directory.
 This means any real ScPCA data or results you had previously downloaded will no longer be in the `data/current` path.
-To switch this symlink path back to ScPCA data or results, rerun either script with the `--release current` option:
+You can use the `--update-symlink` flag, as described below, to update which release directory the `data/current` symlink directs to.
```
I think we need to specify that this option is only part of `download-data.py`?

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -247,46 +265,69 @@ To list all available releases, you can use the following command from the root
-./download-data.py --release current
+## Updating the `current` symlink
+
+As described above, the `data/current` directory is a [symlink](https://en.wikipedia.org/wiki/Symbolic_link) to the most recently-downloaded release directory.
```
```suggestion
As described above, the `data/current` directory is a [symlink](https://en.wikipedia.org/wiki/Symbolic_link) to the most recent release directory.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -223,6 +236,11 @@ data
 ```
 
+!!! tip "The `data/current` directory"
+    The `data/current` directory is a [symlink](https://en.wikipedia.org/wiki/Symbolic_link) to the most recently-downloaded release directory.
````
Trying to make this more accurate. We could add some weasel words here, like "generally" or something about how it might point to test data.
```suggestion
    The `data/current` directory is a [symlink](https://en.wikipedia.org/wiki/Symbolic_link) to the most recent release directory available locally.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -127,6 +137,11 @@ data
 ```
 
+!!! tip "The `data/current` directory"
+    The `data/current` directory is a [symlink](https://en.wikipedia.org/wiki/Symbolic_link) to the most recently-downloaded data release.
````
```suggestion
    The `data/current` directory is a [symlink](https://en.wikipedia.org/wiki/Symbolic_link) to the most recent data release.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
````diff
@@ -247,46 +265,69 @@ To list all available releases, you can use the following command from the root
+We recommend creating a `portal-downloads` subdirectory in the local copy of the `data` directory, which can be accomplished by running the following command from the root of the repository on Linux or macOS:
+
+```sh
+mkdir -p data/portal-downloads
````
I would probably suggest setting up a symlink as well at this point:
```suggestion
mkdir -p data/portal-downloads
# set up a symlink
ln -s data/portal-downloads data/current
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -247,46 +265,69 @@ To list all available releases, you can use the following command from the root
 
--->
\ No newline at end of file
+You can then develop your analysis using these paths.
+
+However, before filing a pull request, you should change your paths to use `data/current` directory [established by the data download script](#download-data-file-structure).
```
If they set up a symlink, they can set their paths up properly the first time.

Which does suggest that we should be explicit about how they should organize files from portal downloads within this directory.

Finally, we should mention that the test data can be downloaded without an account as well.

sjspielman replied:
> Finally, we should mention that the test data can be downloaded without an account as well.

We do have this here, but it sounds like this was not so findable! I'll rework a bit so this is clearer. https://github.com/sjspielman/OpenScPCA-analysis/blob/d0ee8d17b3f2177930d778db0aeb7b804bbb95f3/docs/getting-started/accessing-resources/getting-access-to-data.md?plain=1#L276

jashapiro replied:
Yes, I know it is in there, but I think it makes sense to add a mention here as well as a reminder.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -5,60 +5,48 @@ Here, we describe several ways to access data.
+1. Data from the ScPCA Portal
+    - You can find out more about the contents of files and how they were processed from the ScPCA documentation: <https://scpca.readthedocs.io>.
+    - OpenScPCA project contributors [with an Amazon Web Services (AWS) account](./index.md#getting-access-to-aws) can access data using the provided `download-data.py` script, [as described below](#using-the-download-data-script).
+        - If you have not yet been onboarded to OpenScPCA, you will need to access data [directly from the ScPCA Portal](#accessing-data-from-the-scpca-portal).
```
I don't think this should be a subpoint, and I think it should be phrased positively.
```suggestion
    - All data is also available [directly from the ScPCA Portal](#accessing-data-from-the-scpca-portal) for users who have not yet fully joined the OpenSCPCA project.
```

jashapiro reviewed (commented):
I generally like how things are organized now. I think it is fine to move the portal to the end, but I wanted to make that a bit more prominent in the opening summary.

The other major thing is that the described behavior of `download-data.py` is not quite as you describe: When `--release` is used with any value other than `current` (or `latest`), the symlink is _not_ updated.  The idea is that most of the time you want to just leave `current` alone and somebody downloading an old release is probably doing it for testing purposes, or because they don't want to move to a new release yet (in which case the symlink won't need to be updated).

sjspielman commented:
I'm still working on responding to reviews here, but wanted to highlight this change: https://github.com/AlexsLemonade/OpenScPCA-analysis/pull/573/commits/5e26fc6d1a000daf7365e867753f9c1ed80391d8
In this commit, I updated the section on accessing data from the portal, and I included an example of what a directory structure might look like. But, I realized that the `single_cell_metadata.tsv` scope is different between what the data download script downloads, and what you'd get manually downloading _a single sample_. Which of these do you think makes sense to write then:
- Show structure for a full project download instead
- Don't show this at all
- Something else entirely

jashapiro commented:
> But, I realized that the `single_cell_metadata.tsv` scope is different between what the data download script downloads

I don't think I understand your question? The content of the file will be different, but the location should be the same. It looks to me like you are currently showing a project download, as you include bulk data, which only comes with a project. I think that is the structure you should show.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -318,16 +318,53 @@ Below are some common use cases for this flag:
 mkdir -p data/portal-downloads
+
+# set up a symlink from data/current to data/portal-downloads
+ln -s data/portal-downloads
```
```suggestion
ln -s data/portal-downloads data/current
```

sjspielman commented:
This is the part I was referring to: https://github.com/sjspielman/OpenScPCA-analysis/blob/052652d3baa1a120a6cc6f27db1f9b566b5aa1e0/docs/getting-started/accessing-resources/getting-access-to-data.md?plain=1#L358-L369

The issue is, if they decide to download multiple individual samples from a project but not the full project, each TSV file will contain only that individual sample's metadata. As new ones get downloaded, the TSV would get overwritten with only that fresh sample in it. By contrast, downloading an individual sample via the download script gives you a project-wide TSV no matter what.

jashapiro commented:
> This is the part I was referring to: https://github.com/sjspielman/OpenScPCA-analysis/blob/052652d3baa1a120a6cc6f27db1f9b566b5aa1e0/docs/getting-started/accessing-resources/getting-access-to-data.md?plain=1#L358-L369
> 
> The issue is, if they decide to download multiple individual samples from a project but not the full project, each TSV file will contain only that individual sample's metadata. As new ones get downloaded, the TSV would get overwritten with only that fresh sample in it. By contrast, downloading an individual sample via the download script gives you a project-wide TSV no matter what.

You are assuming they are downloading directly to that folder, which I doubt people will do. If you want, you could make a small note that if they download individual files the metadata file will not include all samples. But that is the most I would do. I don't think it is really an issue for these docs.

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -152,16 +162,11 @@ If you wish to use output from an `OpenScPCA-nf` analysis module in your module,
 !!! tip "Log into AWS CLI before running the script"
-    Before running this script locally, you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
-    To do this, run the following commands in terminal before running the `results-download.py` script, and follow instructions to log in.
+    Before running this script locally (i.e., not from an [ALSF-provided virtual computer](../../aws/index.md#lightsail-for-research-virtual-computing-with-aws)), you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
```
```suggestion
    Before running this script locally (i.e., not using [Lightsail for Research](../../aws/index.md#lightsail-for-research-virtual-computing-with-aws)), you will need to be [logged into your AWS CLI profile](../../technical-setup/environment-setup/configure-aws-cli.md#logging-in-to-a-new-session).
```

sjspielman replied:
Huh, I thought I got this... Oh well, thanks!

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -247,46 +264,93 @@ To list all available releases, you can use the following command from the root
+If you have not yet been [granted access to ScPCA data via AWS](./index.md#getting-access-to-aws), you will need to obtain data directly from [the ScPCA Portal that the Data Lab maintains](https://scpca.alexslemonade.org).
+You can download the project(s) or sample(s) you are interested in analyzing from the ScPCA Portal.
+
+Note that you can also access the [OpenScPCA test data](#accessing-test-data) before getting AWS access as well.
```
duplocated "access" and also/as well
```suggestion
Note that you can download the [OpenScPCA test data](#accessing-test-data) before getting AWS access as well.
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -247,46 +264,93 @@ To list all available releases, you can use the following command from the root
+
+## Accessing data from the ScPCA Portal
+
+If you have not yet been [granted access to ScPCA data via AWS](./index.md#getting-access-to-aws), you will need to obtain data directly from [the ScPCA Portal that the Data Lab maintains](https://scpca.alexslemonade.org).
```
```suggestion
If you have not yet been [granted access to OpenScPCA AWS resources](./index.md#getting-access-to-aws), you will need to obtain data directly from [the ScPCA Portal](https://scpca.alexslemonade.org).
```

jashapiro commented on docs/getting-started/accessing-resources/getting-access-to-data.md:
```diff
@@ -247,46 +264,93 @@ To list all available releases, you can use the following command from the root
+You can then develop your analysis specifying the `data/current` path when reading in data.
+This way, your directory names will be compatible with the file structure established by the [OpenScPCA data download script](#using-the-download-data-script).
+
+Within `data/portal-downloads`, we further recommend organizing your files in the same manner [as the official OpenScPCA data download script does](#downloaded-data-file-structure), as in:
```
```suggestion
Within `data/portal-downloads`, we further recommend organizing your files in the same manner [as the OpenScPCA data download script does](#downloaded-data-file-structure), as in:
```

jashapiro reviewed (approved):
A few small edits but this looks good to go.
